turtle_graphics/turtle.js

```diff
@@ -0,0 +1,148 @@
+class Turtle {
+    constructor(x, y) {
+        this.x = x;
+        this.y = y;
+        this.currentDirection = "E";
+        this.points = [[x, y]];
+        console.log(`starting at ${x},${y}`);
+        this.moveHistory = [];
+        
+        this.moveHistory.push([this.x, this.y]);
+    }
+
+    allPoints(){
+        return this.points;
+    }
+
+
+    forward(steps) {
+        console.log(`^^^^^^^^^^moved forward ${steps} steps in ${this.currentDirection}`);
+
+       console.log(`^^^^^^^^^^currently at ${this.x},${this.y}`);
+        let y1 = this.y;
+        let x1 = this.x;
+
+        switch (this.currentDirection) {
+            case "N":
```

Comment:
All the case values within this statement `case "N"` could be moved into constants and referenced `case NORTH`